src/main/kotlin/app/revanced/patcher/patch/Patch.kt

```diff
@@ -640,19 +656,24 @@ sealed class PatchLoader private constructor(
          *
          * @param binaryClassNamesByPatchesFile The binary class name of the classes to load the patches from
          * associated by the patches file.
+         * @param onLoadPatchesException The callback for patches that could not be loaded.
          *
          * @return The loaded patches associated by the patches file.
          */
-        private fun ClassLoader.loadPatches(binaryClassNamesByPatchesFile: Map<File, Set<String>>) =
-            binaryClassNamesByPatchesFile.mapValues { (_, binaryClassNames) ->
-                binaryClassNames.asSequence().map {
-                    loadClass(it)
-                }.flatMap {
-                    it.patchFields + it.patchMethods
-                }.filter {
-                    it.name != null
-                }.toSet()
-            }
+        private fun ClassLoader.loadPatches(
+            binaryClassNamesByPatchesFile: Map<File, Set<String>>,
+            onLoadPatchesException: (message: String, cause: Throwable) -> Unit
```

Comment:
It would be good to also give the file as well so we can associate the error with the correct bundle in the UI.